.ci/pitest.sh

```diff
@@ -56,6 +55,15 @@ pitest-main)
   checkPitestReport "${ignoredItems[@]}"
   ;;
 
+pitest-coding)
+  mvn -e -P$1 clean test org.pitest:pitest-maven:mutationCoverage;
+  declare -a ignoredItems=(
+  "RedundantThisCheck.java.html:<td class='covered'><pre><span  class='survived'>                    register(ast, frame);</span></pre></td></tr>"
+  "RequireThisCheck.java.html:<td class='covered'><pre><span  class='survived'>                    register(ast, frame);</span></pre></td></tr>"
```

Comment:
This is not ok 
This list of not covered lines is tech debt that  can be only reduced

Reply:
Changes Done.

Reply:
@Rahulkhinchi03 This was undone from latest commit.

Reply:
I have changed it to:
```
  mvn --no-transfer-progress -e -P$1 clean test org.pitest:pitest-maven:mutationCoverage;
  declare -a ignoredItems=(
  "RedundantThisCheck.java.html:<td class='uncovered'><pre><span  class='survived'>                    register(ast, frame);</span></pre></td></tr>"
  "RequireThisCheck.java.html:<td class='uncovered'><pre><span  class='survived'>                    register(ast, frame);</span></pre></td></tr>"
```

Reply:
We are not expecting any new additions to this file at all. As @romani said, it just adds debt and also adds uncertainty if you have enough tests for your changes or if the production code is doing something it probably shouldn't be or shouldn't be needed. This is what pitest shows us.

Reply:
See https://github.com/checkstyle/checkstyle/issues/7797 to learn more about this and pitest.

Reply:
Should I revert my changes for pitest?

Reply:
Yes, after you have tried to resolve the issues locally.

Reply:
How can I improve the Code Coverage issue?

Reply:
You will need to look at the pitest reports and will basically need to figure it out. CI generates these reports so you can find those and print them here and we can look together, unless you want to post your own. We will need to know what mutation is being done to the code and then need to examine what kind of test is needed to show the code is needed.

Reply:
[pitest-coding-coverage-report.zip](https://github.com/checkstyle/checkstyle/files/7747793/pitest-coding-coverage-report.zip)

Reply:
Let me first try to understand and If I will need your help, I will ping you!

Reply:
@rnveach How I can resolve  survive mutations?

Reply:
Open the report and go to the classes you changed. Some lines will be red, others will be green. Look at the red one and try to discern why they are red. There are 2 colors of red, light for missing code coverage and darker for missing mutation coverage. I would start with the light since they should be easier.

Reply:
Do I need to change RequireThisCheck Code too?